### ISSUE TITLE
Cross validation

### ISSUE THREAD

**Issue by maxwellcui** [maxwellcui/4top#12]
A trained model may perform very good on training sample and as well testing sample. The error may increase on unseen data. The reason is that even though the sample space is infinite, only a finite amount of samples are used for the training. The result on the testing sample therefore can't reflect the actual performance on unseen data. A validation set can be separated from the training data. Then the results on the testing set are close to the unseen data.

**Comment by maxwellcui** [comment]
Detail explanation
Case 1: Only training and testing data

|Training model|Err on Testing|Err on unseen data|
|---|---|---|
|Model 1|0.5|>0.5|
|Model 2|0.6|>0.6|
|Model 3|0.4|>0.4|

Case 2: Validation set on training data

|Training model|Err on Validation|Err on Testing| Err on unseen data|
|---|---|---|---|
|Model 1|0.5|>0.5|>0.5|
|Model 2|0.6|>0.6|>0.6|
|Model 3|0.4|>0.4|>0.4|

Case 3: n-fold cross validation

|Set 1 |Set 2|Set 3|Model 1|Model 2|Model 3|
|---|---|---|---|---|---|
|Training|Training|Validation|0.4|0.5|0.6|
|Training|Validation|Training|0.5|0.6|0.8|
|Validation|Training|Training|0.3|0.7|0.9|

|Model number|Average|
|---|---|
|1|0.4|
|2|0.6|
|3|0.7|

**Comment by maxwellcui** [comment]
TMVA provides a `TMVA::CrossValidation` class, but it has its own booking method differing from the `TMVA::Factory` booking method. Code of training needs to be rewritten.
The use of this class needs to be discussed.
